src/sage/schemes/elliptic_curves/ell_curve_isogeny.py

```diff
@@ -3577,12 +3587,18 @@ def compute_isogeny_kernel_polynomial(E1, E2, ell, algorithm=None):
         from sage.misc.superseded import deprecation
         deprecation(34871, 'The "starks" algorithm is being renamed to "stark".')
         algorithm = 'stark'
+
     if algorithm is None:
+        char = E1.base_ring().characteristic()
+        if char != 0 and char < 4*ell+4:
+            raise NotImplementedError('no algorithm for computing kernel polynomial from domain and codomain is implemented for this pair of degree and characteristic')
         algorithm = 'stark' if ell < 10 else 'bmss'
```

Comment:
It would be better to show the degree and the characteristic: `... for degree {ell} and characteristic {char}`.

Please add spaces around `+`.

Otherwise, lgtm.